R/diff_secuTrial.R

```diff
@@ -0,0 +1,58 @@
+#' Returns differences in the setup of two secuTrialdata objects
+#' @description During ongoing studies it is possible that changes to the DataCapture interface
+#'              are made. Sometimes these changes may call for adjustments in analysis code.
+#'              It is considered good practice to run \code{diff_secuTrial()} on the last export
+#'              and the current export of a project to at least make yourself aware of
+#'              potential changes to the setup. If there are differences, the results of this function should
+#'              be interpreted as a first indicator since they may not cover all alterations.
+#'              Information is returned on new forms and variables.
+#'              A detailed list of changes can be produced in the FormBuilder with
+#'              "Compare project setup".
+#' @param x a \code{secuTrialdata} object (the older export)
+#' @param y a \code{secuTrialdata} object (the newer export)
+#' @export
+#' @details If there are differences, \code{diff_secuTrial()} will produce a list of vectors.
+#'          The fist vector informs about new forms and the second vector informs about
+#'          new variables.
+#'
+#' @examples
+#' # export location
+#' export_loc <- system.file("extdata", "sT_exports", "longnames",
+#'                           "s_export_CSV-xls_CTU05_long_ref_miss_en_utf8.zip",
+#'                           package = "secuTrialR")
+#' # read export
+#' sT_export <- read_secuTrial(export_loc)
+#'
+#' # return scores // in this case on the same export
+#' diff_secuTrial(sT_export, sT_export)
```

Comment:
Could you also provide an example where the structure has changed? I think it'd be nice, so that users get an idea of what to expect.

Reply:
I agree 100%. Unfortunately we do not currently have such example data since by design we did not want "things" to change in the example databases. However, you could track an issue for this and I will try and think of a solution when I implement to validation CDMA.

Reply:
Do you still have an old export from one of the databases perhaps... there was a time when one of them didnt have everything so we added more stuff to it... (might even still be in github somewhere)

Reply:
Ok, I will have a look. Thanks for the suggestion.

Reply:
#113

Reply:
So that solution will not work. I tried and the setup tells me "There is already a project with this name. Please enter a different one." - which makes good sense.

We can hold on to issue #113 and I will handle this during the validation process but for now I will leave it as is.

Reply:
Thanks for looking into this.